### PR TITLE
Decompiles z_bg_spot17_bakudankabe.c

### PR DESCRIPTION
**Pull request by i82orbom**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c [hunk 1]
```diff
@@ -21,13 +20,145 @@ const ActorInit Bg_Spot17_Bakudankabe_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot17_Bakudankabe/BgSpot17Bakudankabe_Init.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+extern u32 D_06000A38;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c:31]:
```suggestion
extern UNK_TYPE D_06000A38;
```

File: src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c [hunk 2]
```diff
@@ -21,13 +20,145 @@ const ActorInit Bg_Spot17_Bakudankabe_InitVars = {
...
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void BgSpot17Bakudankabe_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    {
+        u32 gameplayFrames;
+        s8 sp62;
+        s8 sp63;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c:126]:
```suggestion
    u32 gameplayFrames;
    s8 sp62;
    s8 sp63;
    
    {
```

File: src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c [hunk 3]
```diff
@@ -21,13 +20,145 @@ const ActorInit Bg_Spot17_Bakudankabe_InitVars = {
...
+        Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_spot17_bakudankabe.c", 269);
+
+        func_80093D18(globalCtx->state.gfxCtx);
+
+        gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_spot17_bakudankabe.c", 273),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+        sp62 = (sp62 >> 1) + 0xC0;
+        sp63 = (sp63 >> 1) + 0xC0;
+        gameplayFrames = globalCtx->gameplayFrames;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c:143]: What's the purpose of this temp?

File: src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c [hunk 4]
```diff
@@ -21,13 +20,145 @@ const ActorInit Bg_Spot17_Bakudankabe_InitVars = {
...
+        gDPSetEnvColor(gfxCtx->polyOpa.p++, sp62, sp63, 0xFF, 0x80);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c:144]: Color values should be decimal (as of recently)
```suggestion
        gDPSetEnvColor(gfxCtx->polyOpa.p++, sp62, sp63, 255, 128);
```

File: undefined_syms.txt [hunk 5]
```diff
@@ -297,6 +297,12 @@ D_060143A8 = 0x060143A8;
 // z_bg_spot17_funen
 D_06000B40 = 0x06000B40;
 
+// z_bg_spot17_bakudankabe
+D_06000A38 = 0x06000A38;
+D_60008A00 = 0x060008A0;
+D_60009600 = 0x06000960;
```
Comment (Roman971) [inline undefined_syms.txt:303]:
```suggestion
D_060008A0 = 0x060008A0;
D_06000960 = 0x06000960;
```

File: src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c (hunk 3 above)
Comment (i82orbom) [reply src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c:143]: Not needed, converted to pad

**Review by Roman971 — APPROVED** [review]

**Pull request merged**